### PR TITLE
Add data sources for retrieving UC tables

### LINKED ISSUE #1105: [feat] Create data resource to get tables/views within a Unity Catalog database
Currently, to grant permissions for tables and views associated with a **Unity Catalog** a hard-coded list of them has to be maintained in Terraform which is far from ideal. It'd be great to have a data resource like the below which can be then referenced during the grant phase.

```hcl
data "databricks_tables" "db1_tables" {
  catalog_id = databricks_catalog.main.id
  schema_id = databricks_schema.schema.id
}

data "databricks_views" "db1_views" {
  catalog_id = databricks_catalog.main.id
  schema_id = databricks_schema.schema.id
}

resource "databricks_grants" "table_read_grants" {
  for_each = databricks_tables.db1_tables.table_name
  table  = each.key

  grant {
    principal  = "db1_read_user_group"
    privileges = ["SELECT"]
  }
}

resource "databricks_grants" "view_read_grants" {
  for_each = databricks_tables.db1_tables.table_name
  view  = each.key

  grant {
    principal  = "db1_read_user_group"
    privileges = ["SELECT"]
  }
}
```

### PR DESCRIPTION
**Pull request by nfx**
* Added `databricks_catalogs` data
* Added `databricks_schemas` data
* Added `databricks_tables` data

```hcl
data "databricks_schemas" "sandbox" {
  catalog_name = "sandbox"
}

output "all_sandbox_schemas" {
  value = data.databricks_schemas.sandbox
}

data "databricks_tables" "things" {
  catalog_name = "sandbox"
  schema_name = "things"
}

output "all_things_tables" {
  value = data.databricks_tables.things
}

data "databricks_catalogs" "all" {}

output "all_catalogs" {
  value = data.databricks_catalogs.all
}
```

Fixes #1105

### CODE REVIEW

**Comment by nfx** [comment]
Tests will follow next week

File: catalog/data_catalogs.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import (
+	"context"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceCatalogs() *schema.Resource {
+	var data struct {
```
Comment (alexott) [inline catalog/data_catalogs.go:11]: I would better go with map of name to ID or vice versa

File: catalog/data_schemas.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	"context"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceSchemas() *schema.Resource {
+	var data struct {
+		CatalogName string   `json:"catalog_name"`
+		Ids         []string `json:"ids,omitempty" tf:"computed,slice_set"`
```
Comment (alexott) [inline catalog/data_schemas.go:13]: Here as well - map, so we are able to filter by name

File: catalog/data_tables.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceTables() *schema.Resource {
+	var data struct {
+		CatalogName string   `json:"catalog_name"`
+		SchemaName  string   `json:"schema_name"`
+		Ids         []string `json:"ids,omitempty" tf:"computed,slice_set"`
```
Comment (alexott) [inline catalog/data_tables.go:14]: Same here

File: catalog/data_catalogs.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+		Ids []string `json:"ids,omitempty" tf:"computed,slice_set"`
+	}
+	return common.DataResource(&data, func(ctx context.Context, c *common.DatabricksClient) error {
+		catalogsAPI := NewCatalogsAPI(ctx, c)
+		catalogs, err := catalogsAPI.list()
```
Comment (alexott) [inline catalog/data_catalogs.go:16]: How do we select which metastore to query? Or catalog names should be unique across all metastores? I don’t remember

File: catalog/data_catalogs.go (hunk 1 above)
Comment (nfx) [reply catalog/data_catalogs.go:11]: Id of catalog is the name

File: catalog/data_catalogs.go (hunk 4 above)
Comment (nfx) [reply catalog/data_catalogs.go:16]: There's no way to select metastore

**Comment by alexott** [comment]
from code perspective looks ok, will wait for tests

**Comment by vadivelselvaraj** [comment]
@nfx would `databricks_tables` cover views as well?

**Pull request merged**